### PR TITLE
Initial automelt implementation, gui portion.

### PR DESCRIPTION
**Pull request by eamondo2**
Feedback welcome, matching PR open on the main dfhack repo as well.

### CODE REVIEW

**Comment by eamondo2** [comment]
pre-commit.ci autofix

**Review by myk002 — COMMENTED** [review]
I'll need to test, but overall it looks good.

I can see this becoming an interface for all stockpile operations:

automelt
autotrade
stockpiles (copy/import/export)

perhaps we could call this script `gui/stockpiles`, commenting out that script's current functionality until `stockpiles` is working again?

**Comment by myk002** [comment]
linking PRs: DFHack/dfhack#2670

**Review by myk002 — COMMENTED** [review]
needs docs in docs/gui/automelt.rst (you can copy most of the contents from docs/gui/autochop.rst)

**Comment by eamondo2** [comment]
Added docs in `docs/gui/automelt.rst`.

**Comment by eamondo2** [comment]
pre-commit.ci autofix

File: gui/autochop.lua [hunk 1]
```diff
@@ -222,7 +232,7 @@ function Autochop:init()
             visible=is_not_minimal,
         },
         widgets.HotkeyLabel{
-            frame={b=8, l=0},
+            frame={b=7, l=0},
```
Comment (myk002) [inline gui/autochop.lua:235]: to keep the current spacing, could you move these widgets back up one and change the List frame to `b=15`? You can then change the Autochop window frame to `h=28` to compensate.

File: gui/autochop.lua [hunk 2]
```diff
@@ -246,6 +256,26 @@ function Autochop:init()
...
+
+    return false
+end
+
+function Autochop:getDefaultHide()
+
+    if not self.hasEnabledBurrows(self) then
+        return false
+    end
+    return true
```
Comment (myk002) [inline gui/autochop.lua:276]:
```suggestion
    return self:hasEnabledBurrows()
```

File: gui/automelt.lua [hunk 3]
```diff
@@ -195,6 +199,26 @@ function Automelt:init()
...
+
+    return false
+end
+
+function Automelt:getDefaultHide()
+
+    if not self.hasMonitoredStockpiles(self) then
+        return false
+    end
+    return true
```
Comment (myk002) [inline gui/automelt.lua:219]:
```suggestion
    return self:hasMonitoredStockpiles()
```

**Review by myk002 — APPROVED** [review]

**Pull request merged**